crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -1048,7 +1051,46 @@ impl Analyzer<'_, '_> {
 
         matching_elements.dedup_type();
 
-        Ok(Some(Type::union(matching_elements)))
+        let mut res_vec = vec![];
+
+        for el in matching_elements.into_iter() {
+            if let Ok(res) = self.normalize(Some(span), Cow::Owned(el), Default::default()) {
+                res_vec.push(res.into_owned());
+            }
+        }
+        res_vec.dedup_type();
+        if res_vec.len() == 1 {
+            if let Some(ty) = res_vec.pop() {
+                return Ok(Some(self.normalize(Some(span), Cow::Owned(ty), Default::default())?.into_owned()));
+            }
+        }
+        let result = match type_mode {
+            TypeOfMode::LValue => {
+                let mut temp_vec = vec![];
+                for elem in res_vec {
+                    if let Type::Intersection(ty) = elem.normalize() {
+                        for item in ty.types.iter() {
+                            temp_vec.push(item.normalize().clone());
```

Comment:
`normalize().clone()` is an anti-pattern.

Reply:
So how do you take ownership and put it in a vector while melting freezed?

Reply:
You don't need to remove `freezed` state, and you **should never** try to store freezed type as unfreezed form

Reply:
How do you deup if there is a mix of friends saved as frezzled and friends who are not?

Reply:
`dedup_type` works with freezed types without any problem

Reply:
oh.. thx I understand
I fixed their